src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,29 @@
+package duke.tasks;
+
+import duke.exceptions.UserInputException;
+
+import static duke.exceptions.UserInputException.inputExceptionType.EMPTY_DEADLINE_TIME;
+import static duke.exceptions.UserInputException.inputExceptionType.EMPTY_TASK_DESCRIPTION;
+
+//initial skeleton adapted from  https://nus-cs2113-ay2223s2.github.io/website/schedule/week4/project.html partial solution
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String newTaskInfo) throws UserInputException {
+        final String[] deadlineSplit = newTaskInfo.trim().split("/by", 2);
```

Comment:
Both `"/by"` and `2` here can be seen as a magic strings/numbers. You might want to consider using named constants for them to solve the issue. The same applies to some other variables in this and other files too. Sometimes, if applicable, you can also consider using enums to resolve the issue of magic number/string/literal.